app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,23 @@
 require 'uri'
 module Katello
   module KatelloUrlsHelper
+    FOREMAN_SETTING_URL = Setting[:foreman_url].sub(/^https/, 'http').freeze
+
     def subscription_manager_configuration_url(host = nil, rpm = true)
       prefix = if host && host.content_source
-                 "http://#{@host.content_source.hostname}"
+                 "http://#{host.content_source.hostname}"
                else
-                 Setting[:foreman_url].sub(/\Ahttps/, 'http')
+                 FOREMAN_SETTING_URL
```

Comment:
This is not an equivalent to the original code, as it would now need a restart of services for it to apply. Extracting to a separate method, rather than using the constants, would be more appropriate in this case.

Reply:
changed to a method